### DISCUSSION TITLE
does can work on MacBook m1

### DISCUSSION THREAD

**Discussion by cygnus008** [KSP-CKAN/CKAN#3760] [Q&A] [answered]
Started playing Kerbal and am trying to install some mods.  It appears the easiest way is through CKAN. I get everything loaded and have the ckan.dmg and when I click on it nothing happens

**Comment by HebaruSan** [comment]
As far as I know, yes, CKAN works on MacBooks.

https://github.com/KSP-CKAN/CKAN/wiki/Installing-CKAN-on-macOS

**Reply by mattolenik** [reply]
It doesn't work on M1 macs specifically, this should be reopened. It only works on Intel macs.

**Reply by HebaruSan** [reply]
> It doesn't work on M1 macs specifically, this should be reopened. It only works on Intel macs.

Hi @mattolenik, thanks for replying. This is a discussion, not an issue, so it's not possible to close or re-open it. All that can be done is to mark an answer as correct, and as far as I know, this answer is still correct. If you have further information that could change that, please share it at your convenience.

**Reply by mattolenik** [reply]
How is it still correct? It doesn't work on M1 macs because mono fails to run on arm64. Please, I am trying to contribute here, really. It's not great when you _do_ provide new information and then it just gets dismissed. If two people having an issue running on macOS (and there are certainly others), how is it reasonable to just point to a FAQ and say "no, you're wrong"? You said "as far as I know." Well, two people have provided new information...

**Reply by HebaruSan** [reply]
> mono fails to run on arm64

Hi @mattolenik, thanks for mentioning this! I don't have this hardware and so had no way to know about that, and all anyone said up to this point is "doesn't work," which could be due to any number of reasons other than it actually not working. That's exactly what I meant by "further information". :+1: If you'd like to write up an answer to this discussion that covers this, I'd be happy to switch the correct answer marker over to it.

Unfortunately that also puts this problem far outside the scope of anything the CKAN team could address, since we don't maintain Mono. I guess we'll have to add a note in whatever FAQ is applicable here...

**Comment by mattolenik** [comment]
No, it appears broken, I have the same issue as you with my mac on Apple silicon

**Reply by jGus6** [reply]
That's weird-- I've been using CKAN on my 2020 M1 Macbook Air for years with zero problem. I'm using the console UI with Mono 6.12.0.206. I'm running it on iTerm2 (since it's a school computer and Terminal is blocked) and I launch it with a custom Automator script that enters the "mono ckan.exe consoleui" command (again, since terminal is banned). My CKAN version is v1.35.2.24280. I actually found this post because I was wondering why the installation guide says it's incompatible with Apple Silicon. I'm not a dev, but I guess my only suggestion is to upgrade to the newest mono and CKAN versions. Hope you get it working at some point 👍

**Reply by HebaruSan** [reply]
Hi @jGus6, thanks for contributing. The Mono Project's "Supported Platforms" page appears to back up the notion that it won't run on ARM64 Mac systems:

- <https://www.mono-project.com/docs/about-mono/supported-platforms/>

The "Operating system" column only lists "macOS" next to PowerPC, x86, and x86-64, and the "ARM" row says only "Linux (both old and new ABI), iPhone, Android".

Are you by any chance running Mono in some kind of emulation or virtualization environment to work around that limitation?

**Comment by HebaruSan** [comment, marked as answer]
The dependency on Mono was eliminated in #4410, which should fix any issues folks were having with specific platforms. This will be in the next release of CKAN, but can be piloted now with the dev build:

- <https://ksp-ckan.s3-us-west-2.amazonaws.com/CKAN.dmg>

As for some users saying Mono was working for them, I'm guessing they had Rosetta 2 installed and were running it that way.
